### ISSUE TITLE
Feature request: Preseve folder structure

### ISSUE THREAD

**Issue by CarlLee** [ata4/disunity#102]
It would be nice if there's option to preserve folder structure in AssetBundle.
Some games use folder names to link resources.

**Comment by ata4** [comment]
Hm... how is the folder structure not preserved right now? The asset bundle extraction should extract the files exactly as they appear internally in the bundles.

**Comment by CarlLee** [comment]
For example, if a file is in 
      string first = "final/assets/game/cards/07 pe1/part_007/hs3-070_d.psd"

This is a line in AssetBundle/000001.txt extracted from Hearthstone files using ``extract-txt``.
Then, if I use ``extract`` command to retrieve ``hs3-070_d.psd``, it would be stored like ``Texture2D/hs3-070_d.dds``. The important ID number in the folder structure is missing, but that's how Hearthstone retrieve this card textures.

**Comment by mink365** [comment]
yes, folder info is very impotent for many games, so please help to preserve it. Thanks.
